src/components/Button/Button.js

```diff
@@ -2,14 +2,35 @@ import { html } from '../../shared';
 import styles from './Button.module.css';
 
 /**
- * @param {{onClick: (e: MouseEvent) => void, text: string}} props
+ * @param {{ text: string, onClick: (e: MouseEvent) => void, variant?: 'normal' | 'outlined' | 'nextQuestion' | 'gameMode', disabled?: boolean }} props
+ * * @param  {...string} id
```

Comment:
Nie rozumiem tego parametru, zbierze on wszystkie argumenty po pierwszym do tablicy stringów `id`. Element może mieć tylko jedno id plus nie widzę powodu dla którego nie możemy go podać z innymi nazwanymi parametrami w obiekcie.

Reply:
a wiesz co Justyna tak zrobiła i u niej do tego to było potrzebne:
${GameTypes({
  onSelect: (selected) => {
    selectedCategory = selected.id;
  },
  heading: 'Wybierz kategorię',
  categories: [
    { id: 'character', text: 'Co to za postać' },
    { id: 'episode', text: 'Bohaterowie odcinka' },
    { id: 'location', text: 'Kto tu mieszka' },
    { id: 'mixed', text: 'Mieszane' },
  ],
})}

więc po prostu zostawiłem to tak jak ona zrobiła.
@jskrajna Ten parametr id jest potrzebny aby później wygenerować pytania? Czy w sumie można pominąć?

Reply:
tak jak teraz zrobiłem, czyli dodałem osobny parametr do funkcji, do tego jest on opcjonalny. Moim zdaniem najwygodniej tak, czy chodzi o to, że nie jest jako props i nie będzie działać przekazywanie tego id do generowania pytań?

Reply:
jak testowałem to pojawiały się buttony z id, więc moim zdaniem jest okej.

Reply:
Właśnie haczyk polega na tym, że to nie jest [parametr opcjonalny](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters), tylko tzw. [rest parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters) i zmienna `id` będzie przechowywała _tablicę_ stringów a nie stringa. Więc całość obecnie działa, tylko dlatego, że `['test'].toString() === 'test'`.

I zauważ, że będzie działać tylko w tym przypadku (kiedy podamy dokładnie jedno `id`):
- kiedy nie podamy żadnych `id` w buttonie będzie `id=""` co jest niepoprawnym HTMLem.
- kiedy podamy więcej niż jedno `id` (co jest w pełni poprawne w obecnej implementacji: `Button(props, 'id1', 'id2', 'id3')`) dostaniemy `id="id1 id2 id3"` co też jest niepoprawnym HTMLem.

(Chociaż nic się raczej nie popsuje mimo że to niepoprawny HTML, bo podejrzewam, że 95% przeglądarek po prostu zignoruje taki atrybut.)

Szczerze mówiąc też niezbyt ściśle powiedziałem mówiąc "parametr opcjonalny", bo chodziło mi o to, że przy destrukturyzacji obiektu niepodane właściwości dostają wartość `undefined`, bez żadnego błędu. Więc możemy zrobić podobnie jak z `disabled`:
```JavaScript
 * @param {{ text: string, onClick: (e: MouseEvent) => void, variant?: 'normal' | 'outlined' | 'nextQuestion' | 'gameMode', disabled?: boolean, id?: string }} props
```
(znak zapytania po `id` jest żeby wskazać, że to opcjonalne)
Potem:
```JavaScript
function Button({ text, onClick, variant = 'normal', disabled = false, id }) {
```
I `id` będzie stringiem albo ew. `undefined`, więc można z tego skorzystać żeby od razu poprawić ten błąd z niepoprawnym HTMLem:
```JSX
html`<button ${id ? ` id="${id}"` : ''}></button>`
```

Reply:
super rozwiązanie :D woooow 🗡️